src/modules/powerrename/PowerRenameUILib/ExplorerItem.cpp

```diff
@@ -96,18 +121,35 @@ namespace winrt::PowerRenameUI::implementation
             m_checked = value;
             m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"Checked" });
 
-            auto visibility = m_checked && !m_renamed.empty() ? Microsoft::UI::Xaml::Visibility::Visible : Microsoft::UI::Xaml::Visibility::Collapsed;
-            if (m_highlight != visibility)
+            if (m_checked && !m_renamed.empty())
+            {
+                VisualStateManager::GoToState(*this, PowerRenameItemRenameStatusToString(m_state), false);
+            }
+            else
             {
-                m_highlight = visibility;
-                m_propertyChanged(*this, Microsoft::UI::Xaml::Data::PropertyChangedEventArgs{ L"Highlight" });
+                VisualStateManager::GoToState(*this, L"Normal", false);
             }
         }
     }
 
-    Microsoft::UI::Xaml::Visibility ExplorerItem::Highlight()
+    int32_t ExplorerItem::State()
     {
-        return m_highlight;
+        throw hresult_not_implemented();
```

Comment:
Is that intended or just forgotten?

Reply:
It's not a problem, as only a setter was needed and it's not possible to have only a setter defined in IDL. Anyway, I'll fix it to ensure it doesn't break something in the future.